### ISSUE TITLE
ASYNC110 possible false positive when using :=

### ISSUE THREAD

**Issue by alexandrehassan** [astral-sh/ruff#25325]
### Summary

I'm not sure how I could avoid using sleep here

```python
async with asyncio.timeout(timeout):
    while len(result := await do_api_call()) == 0:  # noqa: ASYNC110
        await asyncio.sleep(1)
    return result
```

https://play.ruff.rs/6e0e617d-b34b-4d00-ad55-532ca6e80bc0

### Version

0.15.12, 0.15.14

**Comment by ntBre** [comment]
I don't think I would consider this a false positive. If the rule applies to code like:

```py
async with asyncio.timeout(timeout):
    result = await do_api_call()
    while len(result) == 0:  # noqa: ASYNC110
        await asyncio.sleep(1)
        result = await do_api_call()
    return result
```

Then I think the same [reasoning](https://docs.astral.sh/ruff/rules/async-busy-wait/#why-is-this-bad) from the rule docs applies to the walrus version. Of course, if you're not concerned about the issues described in the docs for this specific case, I think a `noqa` comment is fine.
